File: TranslationFiestaElectron/src/providers/__tests__/providerId.test.ts
import { describe, expect, it } from "vitest";
import { DEFAULT_PROVIDER_ID, normalizeProviderId } from "../../providerId";

describe("providerId", () => {
  it("normalizes aliases to google_unofficial", () => {
    const aliases = [
      "google_unofficial",
      "unofficial",
      "google_unofficial_free",
      "google_free",
      "googletranslate",
      "",
      "  unofficial  ",
      "GOOGLE_UNOFFICIAL",
      undefined,
      null,
      "unknown_provider"
    ];

    for (const alias of aliases) {
      expect(normalizeProviderId(alias)).toBe(DEFAULT_PROVIDER_ID);
    }
  });
});

File: TranslationFiestaFlutter/test/provider_id_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:translation_fiesta_flutter/domain/entities/translation.dart';

void main() {
  test('provider aliases normalize to google_unofficial', () {
    const aliases = <String?>[
      'google_unofficial',
      'unofficial',
      'google_unofficial_free',
      'google_free',
      'googletranslate',
      '',
      '  unofficial  ',
      'GOOGLE_UNOFFICIAL',
      null,
      'unknown_provider',
    ];

    for (final alias in aliases) {
      expect(
        TranslationProviderIdX.fromStorage(alias),
        TranslationProviderId.googleUnofficial,
      );
    }
  });
}
